elab_bridge/server_interface.py

```diff
@@ -63,6 +63,52 @@ def download_experiment(save_to, server_config_json, experiment_id, experiment_a
     return status_code, df
 
 
+def upload_experiment(experiment_file, server_config_json, experiment_title):
+    """
+    Upload an experiment.
+
+    Parameters
+    ----------
+    experiment_file: str
+        Path to the experiment you want to upload. This has to be a json file
+        containing the keys `elabftw' and `extra_fields`.
+    server_config_json: str
+        Path to the json file containing the redcap url, api token and required external modules
+    experiment_title: str
+        The title of the experiment you want to upload
+
+    Returns
+    -------
+        location_response: location of the new template
+    """
+
+    try:
+        with open(experiment_file, 'r') as f:
+            template_json = json.load(f)
+    except json.JSONDecodeError:
+        raise ValueError(f'Invalid JSON file: {experiment_file}')
+
+    if 'extra_fields' not in template_json:
+        raise ValueError('Mandatory field "extra_fields" not present in template')
+
+    with open(experiment_file, 'r') as f:
+        template_form_string = f.read()
+
+    api_client = get_elab_config(server_config_json)
+    experiment_api = elabapi_python.ExperimentsApi(api_client)
+
+    response = experiment_api.post_experiment_with_http_info(body={"category_id": -1})
```

Comment:
Just out of curiosity, why is this "category_id": -1 required here?

Reply:
category_id is The template id to use, or 0 to use the common team template, or -1 to have an empty body.

Given that there is no team template in the sense that the templates are different, I thought that -1 would be the best, no? 

I'm going to add a comment for this line

Reply:
I had a look at the attributes of an experiment and it doesn't seem like it can link to a particular template it was created from. So this is unfortunately not possible :/